app/src/org/commcare/android/database/global/models/ApplicationRecord.java

```diff
@@ -56,4 +84,104 @@ public int getStatus() {
     public void setStatus(int status) {
         this.status = status;
     }
+
+    public String getUniqueId() {
+        return uniqueId;
+    }
+
+    public void setUniqueId(String id) {
+        this.uniqueId = id;
+    }
+
+    public String getDisplayName() {
+        return this.displayName;
+    }
+
+    public void setDisplayName(String appName) {
+        this.displayName = appName;
+    }
+
+    public void setArchiveStatus(boolean b) {
+        this.isArchived = b;
+    }
+
+    public boolean isArchived() {
+        return this.isArchived;
+    }
+
+    @Override
+    public String toString() {
+        return this.displayName;
+    }
+
+    public void setResourcesStatus(boolean b) {
+        this.resourcesValidated = b;
+    }
+
+    public boolean resourcesValidated() {
+        return this.resourcesValidated;
+    }
+
+    /**
+     * Returns true if this ApplicationRecord represents an app generated from
+     * a pre-Multiple Apps version of CommCare that does not have profile files
+     * with uniqueId and displayName
+     */
+    public boolean preMultipleAppsProfile() {
+        return this.preMultipleAppsProfile;
+    }
+    
+    public void setPreMultipleAppsProfile(boolean b) {
+        this.preMultipleAppsProfile = b;
+    }
+
+    /**
+     * Returns true if this ApplicationRecord was just generated from the a
+     * different ApplicationRecord format via the db upgrader, because it was 
+     * initially installed on a phone with a pre-Multiple Apps version of CommCare
+     */
+    public boolean convertedByDbUpgrader() {
+        return this.convertedViaDbUpgrader;
+    }
+
+    public void setConvertedByDbUpgrader(boolean b) {
+        this.convertedViaDbUpgrader = b;
+    }
+
+    /**
+     * Completes a full uninstall of the CommCare app that this ApplicationRecord represents
+     */
+    public void uninstall(Context c) {
+        CommCareApplication._().initializeAppResources(new CommCareApp(this));

```

Comment:
It seems like we might need to unseat the current application (if there is one) before this happens? 

Is it possible for this to fail in weird ways (like there already being a seated app?)